pagopa/io-app
# feat: [IOCOM-498] Payment UI on a PN cancelled message

⚠️ This PR depends on #4970 ⚠️

## Short description
This PR adds support for a single payment on a PN message when the latter is cancelled

| No payment  | Unpaid payment | Paid Payment |
| ------------- | ------------- | - |
| ![00](https://github.com/pagopa/io-app/assets/5150343/e252e5e2-1881-46b8-9ec1-49b5a317d0b7)  | ![01](https://github.com/pagopa/io-app/assets/5150343/8a8d31b5-172c-4d75-a359-752fd31bbaa1)  | ![02](https://github.com/pagopa/io-app/assets/5150343/aa8ec7dc-0e12-4115-bb84-3f60558311b5) |

## List of changes proposed in this pull request
- `ts/features/pn/components/PnMessagePayment`: new component to incapsulate the Payment section on a PN message
- `ts/features/pn/components/PnMessageDetails`: payment verification logic disabled when the notification is cancelled
- `ts/features/pn/components/PnMessageDetails`: UI tracking event moved on its own useEffect in order to be independent from the payment verification logic

## How to test
Make sure to have the local feature flag on for the new PN component. Run the io-dev-api-server with different combination of the cancelled PN message:

- a not cancelled PN message with no payments
- a not cancelled PN message with a valid unpaid payment
- a not cancelled PN message with a valid paid payment
- a cancelled PN message with no payments
- a cancelled PN message with an unpaid payment
- a cancelled PN message with a paid payment

and check that the UI is displayed properly
